docs/modules/architecture/pages/vector-search-overview.adoc

```diff
@@ -0,0 +1,221 @@
+= VectorCollection data structure design
+:description: A Hazelcast vector database engine is a specialized type of database, which is optimized for storing, searching, and managing vector embeddings and additional metadata. You can include values in the metadata to provide filtering, additional processing, or analysis of vectors.
+:page-enterprise: true
+:page-beta: true
+
+{description}
+
```

Comment:
```suggestion

For further information on the design of the Vector Collection data structure, see xref:data-structures:vector-collections.adoc[].
```

Reply:
@shultseva this is the suggested x-ref to tie the topics together (see other comment on nav file)